### PR TITLE
passDogSkill Mutation

### PR DESCRIPTION
**Pull request by Josenecal**
__x__ Wrote Tests
__x__ Implemented
__x__ Reviewed

## Necessary checkmarks:

- [x] All Tests are Passing
- [x] The code will run locally
Notice: One of the tests not related to this PR is not passing locally, but it is on `main` it *is* passing. I assume therefore that this PR will not break that test on `main` but will  just merge the new code into the now fixed code. I've tried to `git pull -r` to rebase my branch with the most recent head, but I keep getting rejected for some reason. 

## Type of change
- [x] New feature
- [] Bug Fix

## Implements/Fixes:

New feature: This implements the passDogSkill endpoint. When provided an ID for a dog_skill, it will update its `passed` status to `true` (or, return an additional warning if it is already `true`) and timestamps the `updated_at` attribute, then returns the dog_skill object with its updated attributes. Or, in the case of a bad id argument, returns a "Bad ID" error.

## Check the correct boxes
- [x] This broke nothing
- [x] All Tests are Passing* (see above)
- [x] The code will run locally

## Testing Changes
- [x] Tests have been added
- [x] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code (I think one line is not tested but I cannot induce a failure that triggers it, it's just a graceful just-in-case failure) 
- [x] I have removed any testing syntax (save_and_open_page/
pry) in my spec files

## Please include a link to a gif of how you feel about this branch:

this-is-how-we-bingham-rollercoaster

### CODE REVIEW

File: spec/factories/dogs.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 FactoryBot.define do
   factory :dog do
     Faker::Config.locale = "en-US"
-    name { Faker::Creature.Dog.name }
-    breed { Faker::Creature.Dog.breed }
-    age { Faker::Number.number.between(from: 1, to: 15) }
+    name { Faker::Creature::Dog.name }
```
Comment (AliciaWatt) [inline spec/factories/dogs.rb:4]: I had changed this too but the fetch dog tests failed with the change. I'm happy with it either way.

**Review by alexGrandolph — COMMENTED** [review]
test looks solid joe!

**Review by alexGrandolph — COMMENTED** [review]
i really like how you handled this error handling. i think it would be worth implementing this style of error handling in our other endpoints

**Review by alexGrandolph — COMMENTED** [review]
adding the argument to the query method is mega smart. love that

**Comment by psitosam** [comment]
Great job. I really like your use of interpolation into the query. I'm going to go refactor one of my specs to use the same thing. :)

**Comment by psitosam** [comment]
We're going to go ahead and merge this, we know which test is failing and why and have fixed it on main, so we'll see if it resolves with the new and improved test or if we have to fix it again. No big deal either way.

**Pull request merged**